test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetAddPackageTests.cs

```diff
@@ -224,5 +224,196 @@ public async Task AddPkg_V3LocalSourceFeed_WithDefaultSolutiuonSource_NoVersionS
                 ridlessTarget.Libraries.Should().Contain(e => e.Version.Equals(packageY_V2.Version));
             }
         }
+
+        [Fact]
+        public async Task AddPkg_WhenPackageNamespacesConfiguredAndNoMatchingSourceFound_Fails()
+        {
+            using var pathContext = new SimpleTestPathContext();
+
+            // Set up solution, and project
+            var solution = new SimpleTestSolutionContext(pathContext.SolutionRoot);
+            var projectName = "projectA";
+            var projectA = XPlatTestUtils.CreateProject(projectName, pathContext, "net5.0");
+
+            const string version = "1.0.0";
+            const string packageX = "X", packageZ = "Z";
+
+            var packageFrameworks = "net472; net5.0";
+            var packageX100 = XPlatTestUtils.CreatePackage(packageX, version, frameworkString: packageFrameworks);
+            var packageZ100 = XPlatTestUtils.CreatePackage(packageZ, version, frameworkString: packageFrameworks);
+
+            packageX100.Dependencies.Add(packageZ100);
+
+            solution.Projects.Add(projectA);
+            solution.Create(pathContext.SolutionRoot);
+
+            var packageSource2 = new DirectoryInfo(Path.Combine(pathContext.WorkingDirectory, "source2"));
+            packageSource2.Create();
+
+            await SimpleTestPackageUtility.CreateFolderFeedV3Async(
+                    pathContext.PackageSource,
+                    PackageSaveMode.Defaultv3,
+                    packageX100,
+                    packageZ100);
+
+            await SimpleTestPackageUtility.CreateFolderFeedV3Async(
+                    packageSource2.FullName,
+                    PackageSaveMode.Defaultv3,
+                    packageX100,
+                    packageZ100);
+
+            var configFile = @$"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+    <packageSources>
+        <add key=""source2"" value=""{packageSource2.FullName}"" />
+    </packageSources>
+        <packageNamespaces>
+            <packageSource key=""source2"">
+                <namespace id=""{packageX}*"" />
+            </packageSource>
+    </packageNamespaces>
+</configuration>
+";
+            var projectADirectory = Path.Combine(pathContext.SolutionRoot, projectA.ProjectName);
+            File.WriteAllText(Path.Combine(projectADirectory, "NuGet.Config"), configFile);
+
+            //Act
+            var result = _fixture.RunDotnet(projectADirectory, $"add {projectA.ProjectPath} package {packageX} -v {version}", ignoreExitCode: true);
+
+            // Assert
+            result.Success.Should().BeFalse(because: result.AllOutput);
+            Assert.Contains($"Installed {packageX} {version} from {packageSource2.FullName}", result.AllOutput);
+            Assert.Contains($"NU1100: Unable to resolve '{packageZ} (>= {version})'", result.AllOutput);
```

Comment:
It is possible to assert the source from `.nupkg.metadata`. I just used this approach because `result.Output` already contains that information and we can avoid another IO operation to read from `metadata` file.